doc/sflow/sflow_hld.md

```diff
@@ -0,0 +1,456 @@
+     */
+    SAI_HOSTIF_ATTR_NAME,
+
+ /**
+     * @brief Set the Generic netlink (multicast) port id on which the packets/buffers
+     * are received on this host interface
+     *
+     * @type sai_uint32_t
+     * @flags CREATE_AND_SET
+     * @default 0
+     */
+    SAI_HOSTIF_ATTR_GENETLINK_PORT_ID,
+
+ /** Receive packets via Linux generic netlink interface */
+    SAI_HOSTIF_TABLE_ENTRY_CHANNEL_TYPE_GENETLINK
+```
+#### Creating a GENETLINK Host Interface
+
+Below is an example code snip that shows how a GENETLINK based host inerface is created. It is assumed that the application has already installed the psample.ko and created multicast group 100.
+
+```
+// Create a Host Interface based on generic netlink
+sai_object_id_t host_if_id;
+sai_attribute_t sai_host_if_attr[3];
+ 
+sai_host_if_attr[0].id=SAI_HOST_INTERFACE_ATTR_TYPE;
+sai_host_if_attr[0].value=SAI_HOST_INTERFACE_TYPE_GENETLINK;
+ 
+sai_host_if_attr[1].id= SAI_HOST_INTERFACE_ATTR_NAME;
+sai_host_if_attr[1].value=”psample”;
+ 
+sai_host_if_attr[2].id= SAI_HOSTIF_ATTR_GENETLINK_PORT_ID;
+sai_host_if_attr[2].value=100;
+
+sai_create_host_interface_fn(&host_if_id, 9, sai_host_if_attr);
+```
+
+### Mapping a sFlow (SAI_HOSTIF_TRAP_TYPE_SAMPLEPACKET) trap to a GENETLINK host interface multicast group id
+
+Below is the code snip that outlines how an sFlow trap is mapped to the GENETLINK host interface created in the previous section.
+
+```
+// Configure the host table to receive traps on the generic netlink socket
+
+sai_object_id_t host_table_entry;
+sai_attribute_t sai_host_table_attr[9];
+ 
+sai_host_table_attr[0].id=SAI_HOSTIF_TABLE_ENTRY_ATTR_TYPE;
+sai_host_table_attr[0].value= SAI_HOST_INTERFACE_TABLE_ENTRY_TYPE_TRAP_ID;
+ 
+sai_host_table_attr[1].id= SAI_HOSTIF_TABLE_ENTRY_ATTR_TRAP_ID;
+sai_host_table_attr[1].value=sflow_trap_id; // Object referencing SAMPLEPACKET trap
+
+sai_host_table_attr[2].id= SAI_HOSTIF_TABLE_ENTRY_ATTR_CHANNEL;
+sai_host_table_attr[2].value=  SAI_HOSTIF_TABLE_ENTRY_CHANNEL_TYPE_GENETLINK; 
+
+sai_host_table_attr[3].id= SAI_HOSTIF_TABLE_ENTRY_ATTR_HOST_IF;
+sai_host_table_attr[3].value=host_if_id; // host interface of type file descriptor for GENETLINK
+
+sai_create_hostif_table_entry_fn(&host_table_entry, 4, sai_host_table_attr);
+```
+
+It is assumed that the trap group and the trap itself have been defined using sai_create_hostif_trap_group_fn() and sai_create_hostif_trap_fn().
+
+## 7 **sFlow in Virtual Switch**
+
+On the SONiC VS, SAI calls would map to the tc_sample commands on the switchport interfaces (http://man7.org/linux/man-pages/man8/tc-sample.8.html).
+
+## 8 **Build**
+
+* The host-sflow package will be built with the mod_sonic callback implementations using the FEATURES="SONIC" option
+
+## 9 **Restrictions**
+* /etc/hsflowd.conf should not be modified manually. While it should be possible to change /etc/hsflowd.conf manually and restart the sflow container, it is not recommended.

```

Comment:
is there any hsflowd/InMon license implications? Community interested to learn on the implications if any.

Reply:
https://github.com/sflow/host-sflow/blob/master/LICENSE
https://inmon.com/technology/sflowlicense.txt